namerd/dcos-bootstrap/src/main/scala/com/twitter/finagle/serverset2/ZkClient.scala

```diff
@@ -0,0 +1,62 @@
+package com.twitter.finagle.serverset2
+
+import com.twitter.finagle.Path
+import com.twitter.finagle.serverset2.client.{ClientBuilder, CreateMode, Data, KeeperException, SessionState, WatchState, ZooKeeperWriter}
+import com.twitter.io.Buf
+import com.twitter.util.{Duration, Future}
+
+/**
+ * A generic Zk Client
+ */
+class ZkClient(
+  hosts: String,
+  zkPrefix: String,
+  sessionTimeout: Option[Duration]
+) {
+  private[this] val builder = ClientBuilder()
+    .hosts(hosts)
+    .sessionTimeout(sessionTimeout.getOrElse(ZkSession.DefaultSessionTimeout))
+
+  def create(ns: String, data: Buf): Future[Unit] = {
+    writerConnect().flatMap { zkw =>
+      ensurePath(zkw, zkPrefix).flatMap { _ =>
+        zkw.create(
+          s"$zkPrefix/$ns",
+          Some(data),
+          Seq(Data.ACL.AnyoneAllUnsafe),
+          CreateMode.Persistent
+        ).rescue {
+            case KeeperException.NodeExists(_) => Future.Unit
+          }
+      }
+    }.unit
+  }
+
+  private[this] def ensurePath(zkw: ZooKeeperWriter, path: String): Future[Unit] = {
+    val components = Path.read(path)
```

Comment:
I think this ties together the syntax for finagle Paths and the syntax for zookeeper paths.  They're only the same by coincidence.  What about just `path.split("/")`?